coral-incremental/src/main/java/com/linkedin/coral/incremental/RelIncrementalTransformer.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.incremental;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.calcite.plan.RelOptTable;
+import org.apache.calcite.prepare.RelOptTableImpl;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.RelShuttle;
+import org.apache.calcite.rel.RelShuttleImpl;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.logical.LogicalAggregate;
+import org.apache.calcite.rel.logical.LogicalFilter;
+import org.apache.calcite.rel.logical.LogicalJoin;
+import org.apache.calcite.rel.logical.LogicalProject;
+import org.apache.calcite.rel.logical.LogicalTableScan;
+import org.apache.calcite.rel.logical.LogicalUnion;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexNode;
+
+
+public class RelIncrementalTransformer {
```

Comment:
`RelNodeIncrementalTransfomer` for consistency with the visualizer (interesting co-incidence).

Reply:
Can this class extend the `RelShuttleImpl`?

Reply:
Hmm, I'm not sure that it makes sense for the entire transformer class to be extending `RelShuttleImpl`. For example, something like `IRRelToSparkRelTransformer`, which the format of this class was based off of, does not extend `RelShuttleImpl`.